### PR TITLE
fix all Framework imports that would break cocoapods installation

### PR DESCRIPTION
**Pull request by fcerdeiral**
This PR will fix cocoapods integration for the libraries `appcenter`, `appcenter-crashes` and `appcenter-push`. They're currently failing due to wrong use of `@import` statements inside its headers and implementation files.  These libraries should build without the need to import `AppCenter` directly, just by knowing theirs `.h` files.

The issue can be demonstrated if you try to integrate using purely cocoapods and not by adding the xcode project files inside your project. If done so you should get an error like this one:
```
▸ Compiling AppCenterReactNative.m

AppCenterReactNativeShared/AppCenterReactNativeShared/AppCenterReactNativeShared.h:3:9: module 'AppCenter' not found
```

This change removes that import from the header file and replaces it with a Forward Declaration of the class or protocol if needed. Then inside the implementation file the needed header is imported using `#import <Framework/Header.h>` format instead of the module format, that should work for all cases.

### CODE REVIEW

File: appcenter-push/ios/AppCenterReactNativePush/AppCenterReactNativePushDelegate.h [hunk 1]
```diff
@@ -11,6 +9,7 @@
 #endif
 
 @class AppCenterReactNativePush;
+@protocol MSPushDelegate;
```
[the remark this answers is not in the document]
Comment (fcerdeiral) [reply appcenter-push/ios/AppCenterReactNativePush/AppCenterReactNativePushDelegate.h]: ✅

**Comment by fcerdeiral** [comment]
That's odd because here it's all normal. I've run the script inside the `TestApp` called `update-link-to-appcenter.sh`, then opened Xcode and it built everything without any issues on my machine. Check this screenshot that specifically shows the compile log of this file.
screen shot 2018-05-30 at 4 35 20 pm

The only two things that I noticed were:

- The header inside the `/AppCenterReactNativeShared.framework` was not updated (just committed that)
- And that Xcode produces a lot of changes after linking with that script, but I believe those are just a bunch of IDs that change inside `.xcodeproj` file on linking and will change every time.

Please let me know if you have more documentation on how to run all the tests you guys run on a PR. I might have missed something in the process. I didn't find docs for contributing.

**Comment by bmourat** [comment]
That's strange now everything compiles, maybe that is because I was performing all those operations manually and missed something.

**Comment by fcerdeiral** [comment]
@bmourat hey.. any idea of when this will be merged into `develop` or `master`? We're currently blocked on this to make AppCenter integration work on iOS. Thanks!

**Comment by guperrot** [comment]
Hi, we release our SDKs the third week of the month. If you need packages before you can run `scripts/npm-pack.sh` on your branch and use those locally or in a private or non official repo. As for merging we will do it soon but release will come later after the merge.

**Comment by fcerdeiral** [comment]
Great! thanks for the quick response @guperrot

**Comment by dhei** [comment]
Hey @fcerdeiral, thanks for the contribution, your PR is merged and will be shipped in our June SDK release.

**Pull request merged**